### ISSUE TITLE
no-empty-class doesn't respect escape

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/eslint#289]
The following regular expression will be flagged as error by no-empty-class:

```
/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g
```

I have no idea why, but the following RegEx is also flagged by the same rule:

```
/\s*:\s*/g
```
